### PR TITLE
Firewall (iptables/ufw) / cron changes: multiconfiguration; automatic CIDR calculation

### PR DESCRIPTION
**Pull request by dualmi**
- Automatic netmask conversion in CIDR format with ansible instead of hardcoded /24
- Multiconfiguration firewall and cron fixes
- Client list sorting fix for sync certificates

### CODE REVIEW

File: tasks/client_keys.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@
 - name: Fetch client config
   fetch:
     src: "{{ openvpn_ovpn_dir }}/{{ item }}-{{ inventory_hostname }}.ovpn"
-    dest: "{{ openvpn_fetch_client_configs_dir }}/{{ item }}/{{ inventory_hostname }}{{ openvpn_fetch_client_configs_suffix }}.ovpn"
+    dest: "{{ openvpn_fetch_client_configs_dir }}/{{ item }}-{{ inventory_hostname }}{{ openvpn_fetch_client_configs_suffix }}.ovpn"
```
Comment (kyl191) [inline tasks/client_keys.yml:80]: Why is this changed? 
This will put all the configs in a single folder instead of having a directory per client.

I don't really want to change it for everyone using this without justification.

File: README.md [hunk 2]
```diff
@@ -186,6 +186,21 @@ Does not depend on any other roles
 
 > **Note:** As the role will need to know the remote used platform (32 or 64 bits), you must set `gather_facts` to `true` in your play.
 
+## Requirements for MikroTik clients
+MikroTik routers (https://mt.lv) doesn't support some OpenVPN features: UDP mode, LZO compression, TLS authentication (according to https://wiki.mikrotik.com/wiki/Manual:Interface/OVPN).
+
+So here is example which additional variables you may have in playbook if your clients will be Mikrotik routers (also if you already have standart udp configuration, create a separate config file with variables below and this config wouldn't affect working users):
```
Comment (kyl191) [inline README.md:192]: Typo in `standart` (I think you mean standar*d*)

File: tasks/cert_sync_detection.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@
   shell:
     cmd: |
       set -o pipefail
-      ls -1 *.csr | { grep -v server.csr || true; } | sort | cut -f1 -d'.' > {{ tempfile_existing.path }}
+      ls -1 *.csr | { grep -v server.csr || true; } | cut -f1 -d'.' | sort > {{ tempfile_existing.path }}
```
Comment (kyl191) [inline tasks/cert_sync_detection.yml:25]: Do you have an example of how this broke?

I see your commit mentioning non-lexicographic sorting, which is kind of weird given that it's supposed to be taking the first field before the `.`, which afaik sort did by default.

File: tasks/server_keys.yml [hunk 4]
```diff
@@ -154,14 +154,14 @@
 - name: Install crl-cron script
   template:
     src: crl-cron.sh.j2
-    dest: "{{ openvpn_base_dir }}/crl-cron.sh"
+    dest: "{{ openvpn_base_dir }}/crl-cron_{{ openvpn_config_file }}.sh"
```
Comment (kyl191) [inline tasks/server_keys.yml:157]: I'm not sure why you need to change this.

What is your use case?

**Review by kyl191 — CHANGES_REQUESTED** [review]
openvpn_server_netmask_cidr  and s/tun0/tun+ changes are fine.

I'd like to understand more about changing the ovpn files & why the crl cron files need to be per-ovpn server if you're not creating a CA per server.

File: tasks/server_keys.yml (hunk 4 above)
Comment (keir-ru) [reply tasks/server_keys.yml:157]: When `openvpn_key_dir` variable is set, this file contains different path for each openvpn instance.

My case is - few playbooks with different users/routes, different `openvpn_server_network`, `openvpn_server_netmask`, `openvpn_key_dir`, `openvpn_port` and other variables applied to one server.

File: tasks/cert_sync_detection.yml (hunk 3 above)
Comment (keir-ru) [reply tasks/cert_sync_detection.yml:25]: It will affects you when you have user list sorted like this:
```
auser
buser
buser-mobile
cuser
```

File sorting will be different (which comm util cannot handle)

example:
```
~ $ touch auser.csr buser.csr buser-mobile.csr cuser.csr
~ $ ls -1 *.csr | sort | cut -f1 -d'.'
auser
buser-mobile
buser
cuser
~ $ ls -1 *.csr | cut -f1 -d'.' | sort
auser
buser
buser-mobile
cuser
```

File: README.md (hunk 2 above)
Comment (keir-ru) [reply README.md:192]:
```suggestion
So here is example which additional variables you may have in playbook if your clients will be Mikrotik routers (also if you already have standard udp configuration, create a separate config file with variables below and this config wouldn't affect working users):
```

File: tasks/client_keys.yml (hunk 1 above)
Comment (keir-ru) [reply tasks/client_keys.yml:80]: I made this change firstly for myself because whole file name contains user name after this - more usable when you sending .ovpn profile file to user after creation. Github automatically applied this commit to same PR, thats why it here. Of course this is not necessary, hope you can decline 1 commit (can you?)

Comment (kyl191) [reply tasks/client_keys.yml:80]: Nope, it's not giving me the opportunity to undo individual commits.

Just do a commit over it?

File: tasks/server_keys.yml (hunk 4 above)
Comment (kyl191) [reply tasks/server_keys.yml:157]: So multiple OpenVPN servers on the same server. Hadn't thought of that.

How about you just change it to use the `{{ openvpn_key_dir }}` instead of renaming it within `{{ openvpn_base_dir }}`, then update the cron line that references it: https://github.com/kyl191/ansible-role-openvpn/blob/master/tasks/server_keys.yml#L181
